src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/OtlpProtobufSerializer.cs

```diff
@@ -204,7 +204,54 @@ private void SerializeMetric(byte[] buffer, ref int cursor, Metric metric)
             case MetricType.DoubleSum:
             case MetricType.DoubleSumNonMonotonic:
                 {
-                    // TODO
+                    cursor = this.instrumentValueIndex;
+
+                    // Write isMonotonic tag
+                    ProtobufSerializerHelper.WriteBoolWithTag(buffer, ref cursor, FieldNumberConstants.Sum_is_monotonic, metric.MetricType == MetricType.DoubleSum);
+
+                    // Write aggregationTemporality tag
+                    ProtobufSerializerHelper.WriteEnumWithTag(buffer, ref cursor, FieldNumberConstants.Sum_aggregation_temporality, metric.Temporality == AggregationTemporality.Cumulative ? 2 : 1);
+
+                    this.metricPointTagAndLengthIndex = cursor;
+                    this.metricPointValueIndex = cursor + TagAndLengthSize;
+                    foreach (var metricPoint in metric.GetMetricPoints())
+                    {
+                        try
+                        {
+                            // Reset cursor to write new metricPoint
+                            cursor = this.metricPointValueIndex;
+
+                            var sum = metricPoint.GetSumDouble();
+
+                            ProtobufSerializerHelper.WriteDoubleWithTag(buffer, ref cursor, FieldNumberConstants.NumberDataPoint_as_double, sum);
```

Comment:
Consider moving the serialization of `NumberDataPoint` to another method so that it can be reused as it would be required for all `Sum` and `Gauge` types.

Reply:
I did try this earlier, it was not helping much due to types involved and the way we get the sum component for different types. Let me give it another try when I add the gauge types.

Reply:
There is probably some clean way to do it with generics.

Reply:
Good point, I will explore this.